test/Microsoft.AspNetCore.SignalR.Common.Tests/Internal/Protocol/HandshakeProtocolTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Common.Tests.Internal.Protocol
+{
+    public class HandshakeProtocolTests
+    {
+        [Fact]
+        public void CanRoundtripHandshakeRequest()
```

Comment:
I know the old tests for this did the same thing, but I'm not a fan of "roundtrip" tests for our protocols since we have a JS client as well (and possibly other clients in the future). Can we make these proper "given this specific JSON I get this parsed object" and "given this specific object I get this JSON" tests rather than using roundtripping? There's not much variety here, so it should be pretty straightforward.

Round-tripping tests don't break if we break the protocol in both the reader and the writer.